### PR TITLE
Change Dependency to members

### PR DESCRIPTION
**Pull request by ivanblinov2k17**

### CODE REVIEW

File: packages/inferno-generator/src/expressions/inferno-component.ts [hunk 1]
```diff
@@ -193,20 +193,22 @@ export class InfernoComponent extends PreactComponent {
...
     if (this.effects.length || this.jQueryRegistered) {
-      const dependencies = this.effects.map(
+      const dependenciesString = this.effects.map(
         (e) => e.getDependency(this.getToStringOptions())
-          .filter((dep) => {
-            const depMember = this.getToStringOptions().members.find(
-              (member) => member.name === dep,
-            );
-            return !depMember?.isMutable;
+          .filter((dep) => (dep instanceof BaseClassMember ? !dep.isMutable : true))
```
Comment (MikeVitik) [inline packages/inferno-generator/src/expressions/inferno-component.ts:198]: I suggest to remove the filter and move !dep.isMutable in to the map if expression

File: packages/react-generator/src/expressions/class-members/get-accessor.ts [hunk 2]
```diff
@@ -9,10 +10,14 @@ export class GetAccessor extends BaseGetAccessor {
...
   }
 
-  getDependency(options: toStringOptions): string[] {
-    return calculateMethodDependency(
+  getDependencyString(options: toStringOptions): string[] {
+    const dependencies = calculateMethodDependency(
       super.getDependency(options),
       options.members,
     );
+    return dependencies.reduce((arr: string[], dep) => (dep instanceof BaseClassMember
```
Comment (MikeVitik) [inline packages/react-generator/src/expressions/class-members/get-accessor.ts:18]: For me the next is more clear:
```
 [
...arr, 
...(dep instanceof BaseClassMember ? dep.getDependencyString(options) : dep)
])
```

File: packages/react-generator/src/expressions/class-members/method.ts [hunk 3]
```diff
@@ -19,14 +21,18 @@ export function calculateMethodDependency(
...
 
-  getDependency(options: toStringOptions) {
-    return calculateMethodDependency(
+  getDependencyString(options: toStringOptions): string[] {
+    const dependencies = calculateMethodDependency(
       super.getDependency(options),
       options.members,
-    );
+    ).filter((dep) => dep !== this);
+    return dependencies.reduce((arr: string[], dep) => (dep instanceof BaseClassMember
```
Comment (MikeVitik) [inline packages/react-generator/src/expressions/class-members/method.ts:33]: See prev comment

File: packages/react-generator/src/expressions/react-component.ts [hunk 4]
```diff
@@ -456,16 +457,34 @@ export class ReactComponent extends Component {
...
+            });
+            if (deps.indexOf('props') > -1) {
+              deps = deps.filter(
+                (d) => !(d instanceof BaseClassMember
+                    && getProps(this.members).includes(d as Property))
+                  || d._hasDecorator(Decorators.TwoWay),
+              );
+            }
+            const depNames = deps.reduce((arr: string[], dep) => {
+              if (dep instanceof BaseClassMember) {
```
Comment (MikeVitik) [inline packages/react-generator/src/expressions/react-component.ts:479]: see prev comment

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: packages/react-generator/src/expressions/class-members/get-accessor.ts (hunk 2 above)
Comment (ivanblinov2k17) [reply packages/react-generator/src/expressions/class-members/get-accessor.ts]: if we will do such way, then if dep = 'str' will return 's', 't', 'r'

**Pull request merged**